src/olympia/versions/models.py

```diff
@@ -168,6 +169,33 @@ def auto_approvable(self):
         )
         return qs
 
+    def should_have_due_date(self, not_=False):
+        return getattr(self, 'exclude' if not_ else 'filter')(
+            Q(file__status=amo.STATUS_AWAITING_REVIEW)
+            & Q(reviewerflags__pending_rejection__isnull=True)
+            & Q(
+                Q(addon__type__in=amo.GROUP_TYPE_THEME)
+                | Q(
+                    addon__promotedaddon__group_id__in=(g.id for g in PRE_REVIEW_GROUPS)
+                )
```

Comment:
We currently ignore promotions for unlisted (which allows pseudo-beta versions to be auto-approval, so I assume we want to continue doing that).

Reply:
I can see pros and cons with that but if that's what we currently do then I'll keep it the same

Reply:
https://github.com/mozilla/addons-server/pull/20184/commits/15ac129c2b2179e5aec0fd93ecc76cf8815461dd